src/fabric/src/fabric_db_purged_infos.erl

```diff
@@ -0,0 +1,88 @@
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(fabric_db_purged_infos).
+
+-export([go/1]).
+
+-include_lib("fabric/include/fabric.hrl").
+-include_lib("mem3/include/mem3.hrl").
+
+-record(pacc, {
+    counters,
+    replies,
+    ring_opts
+}).
+
+go(DbName) ->
+    Shards = mem3:shards(DbName),
+    Workers = fabric_util:submit_jobs(Shards, get_purged_infos, []),
+    RexiMon = fabric_util:create_monitors(Shards),
+    Fun = fun handle_message/3,
+    Acc0 = #pacc{
+        counters = fabric_dict:init(Workers, nil),
+        replies = [],
+        ring_opts = [{any, Shards}]
+    },
+    try
+        case fabric_util:recv(Workers, #shard.ref, Fun, Acc0) of
+            {ok, Res} ->
+                {ok, Res};
+            {timeout, {WorkersDict, _}} ->
+                DefunctWorkers = fabric_util:remove_done_workers(WorkersDict, nil),
+                fabric_util:log_timeout(DefunctWorkers, "get_purged_infos"),
+                {error, timeout};
+            {error, Error} ->
+                throw(Error)
+        end
+    after
+        rexi_monitor:stop(RexiMon)
+    end.
+
+handle_message({rexi_DOWN, _, {_, NodeRef}, _}, _Shard, #pacc{} = Acc) ->
+    #pacc{counters = Counters, ring_opts = RingOpts} = Acc,
+    case fabric_util:remove_down_workers(Counters, NodeRef, RingOpts) of
+        {ok, NewCounters} ->
+            {ok, Acc#pacc{counters = NewCounters}};
+        error ->
+            {error, {nodedown, <<"progress not possible">>}}
+    end;
+handle_message({rexi_EXIT, Reason}, Shard, #pacc{} = Acc) ->
+    #pacc{counters = Counters, ring_opts = RingOpts} = Acc,
+    NewCounters = fabric_dict:erase(Shard, Counters),
+    case fabric_ring:is_progress_possible(NewCounters, RingOpts) of
+        true ->
+            {ok, Acc#pacc{counters = NewCounters}};
+        false ->
+            {error, Reason}
+    end;
+handle_message({ok, Info}, #shard{dbname = Name} = Shard, #pacc{} = Acc) ->
+    #pacc{counters = Counters, replies = Replies} = Acc,
+    Replies1 = [Info | Replies],
+    Counters1 = fabric_dict:erase(Shard, Counters),
+    case fabric_dict:size(Counters1) =:= 0 of
+        true ->
+            {stop, [{purged_infos, format_purged_infos(Replies1)}]};
+        false ->
+            {ok, Acc#pacc{counters = Counters1, replies = Replies1}}
+    end;
+handle_message(_, _, #pacc{} = Acc) ->
+    {ok, Acc}.
+
+format_purged_infos(Info) ->
+    PurgedIdRevs = lists:flatten(Info),
+    Ids = proplists:get_keys(PurgedIdRevs),
+    PurgeInfoRaw = [{Id, couch_lists:uniq(proplists:append_values(Id, PurgedIdRevs))} || Id <- Ids],
+    lists:sort([
+        {[{id, Id}, {revs, [couch_doc:rev_to_str(Rev) || Rev <- Revs]}]}
+     || {Id, Revs} <- PurgeInfoRaw
+    ]).
```

Comment:
What do you think of returning the exact same representation as used by the `_purge` request: `{ $id : [$rev1, $rev2, ...], ...}`. That could allow easily piping the requests directly from a GET result to POST `_purge` on another cluster?

Reply:
Good idea! Changed and thanks for the review!